src/libraries/System.Private.CoreLib/src/System/SpanHelpers.T.cs

```diff
@@ -1342,5 +1177,1140 @@ public static int SequenceCompareTo<T>(ref T first, int firstLength, ref T secon
             }
             return firstLength.CompareTo(secondLength);
         }
+
+        [MethodImpl(MethodImplOptions.AggressiveOptimization)]
+        internal static bool ContainsValueType<T>(ref T searchSpace, T value, int length) where T : struct, INumber<T>
```

Comment:
Why is `INumber` better than `IEquatable` here? `IEquatable` seemed to be more appropriate.

Reply:
For `IndexOfAny` for two and more values, the codegen is better for `==` rather than `.Equals`.

Also, it's more restrictive (fewer types meet the criteria)

Reply:
> the codegen is better for == rather than .Equals.

Sounds like a JIT code quality issue that should be fixed.

The downsides are:
- `Equals` and `INumber<T>.operator==` are not semantically the same. They happen to have the same semantics for the types used here, but it is not generally the case. For example, it would not be the case for `float` or `double`. Maybe these  methods should be called `ContainsNumber`, etc. to make the semantics difference more obvious?
- `INumber<T>` is more heavy weight interface. Using it here is likely going to add several kBs of footprint to trimmed apps benchmarks (if they are not using it anywhere else yet). It may be interesting to measure it.